Submodule src/Infrastructure contains modified content
diff --git a/src/Infrastructure/src/Emulator/Cores/linux-properties.csproj b/src/Infrastructure/src/Emulator/Cores/linux-properties.csproj
index c1639144..629486f0 100644
--- a/src/Infrastructure/src/Emulator/Cores/linux-properties.csproj
+++ b/src/Infrastructure/src/Emulator/Cores/linux-properties.csproj
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
 <PropertyGroup>
-  <CompilerPath>/usr/bin/gcc</CompilerPath>
-  <LinkerPath>/usr/bin/gcc</LinkerPath>
-  <ArPath>/usr/bin/ar</ArPath>
+  <CompilerPath>$(CC)</CompilerPath>
+  <LinkerPath>$(CC)</LinkerPath>
+  <ArPath>$(AR)</ArPath>
   <CurrentPlatform>Linux</CurrentPlatform>
   <DefineConstants>$(DefineConstants);PLATFORM_LINUX</DefineConstants>
 </PropertyGroup>
diff --git a/src/Infrastructure/src/Emulator/Cores/osx-properties.csproj b/src/Infrastructure/src/Emulator/Cores/osx-properties.csproj
index cc4ac2a2..7592904b 100644
--- a/src/Infrastructure/src/Emulator/Cores/osx-properties.csproj
+++ b/src/Infrastructure/src/Emulator/Cores/osx-properties.csproj
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
 <PropertyGroup>
-  <CompilerPath>/usr/bin/clang</CompilerPath>
-  <LinkerPath>/usr/bin/clang</LinkerPath>
-  <ArPath>/usr/bin/ar</ArPath>
+  <CompilerPath>$(CC)</CompilerPath>
+  <LinkerPath>$(CC)</LinkerPath>
+  <ArPath>$(AR)</ArPath>
   <CurrentPlatform>OSX</CurrentPlatform>
   <DefineConstants>$(DefineConstants);PLATFORM_OSX</DefineConstants>
 </PropertyGroup>
diff --git a/tools/building/check_weak_implementations.sh b/tools/building/check_weak_implementations.sh
index 5e4cd50..8df5623 100755
--- a/tools/building/check_weak_implementations.sh
+++ b/tools/building/check_weak_implementations.sh
@@ -27,7 +27,7 @@ for PATH_WEAK in $PATHS_WEAK
 do
   PATH_IMPLEMENTATION=`first $PATHS_IMPLEMENTATION`
   PATHS_IMPLEMENTATION=`next $PATHS_IMPLEMENTATION`
-  gcc -I tlib/include -E $PATH_WEAK | grep weak | grep -o tlib[_A-Za-z]* | sort | uniq > $WEAKS
+  $CC -I tlib/include -E $PATH_WEAK | grep weak | grep -o tlib[_A-Za-z]* | sort | uniq > $WEAKS
   cat $PATH_IMPLEMENTATION | grep -o tlib[_A-Za-z]* | sort | uniq > $IMPLEMENTATIONS
   if grep -vwF -f $IMPLEMENTATIONS $WEAKS
   then
